### DISCUSSION TITLE
Webp support on Windows

### DISCUSSION THREAD

**Discussion by JunglePredator** [pdfarranger/pdfarranger#975] [Q&A]
I don't know if it's officially supported by the PDF format but it'd be nice if you added Webp support.
Tested PDFs containing Webp with Foxit & Sumatra & both readers were fine with it.

**Comment by jeromerobert** [comment]
webp is not supported by PDF. pdfarranger can import and convert webp files thanks to PIL. If it does not work I guess that you have the same issue as https://github.com/pdfarranger/pdfarranger/issues/937. Please do the same as in #937.

See https://github.com/pdfarranger/pdfarranger/issues/937#issuecomment-1722255410 and test with webp instead of jpg.

**Reply by JunglePredator** [reply]
While I do get none/none from the Python command, I don't believe it's the same problem.
WebP opens & is supported on my computer fine (using XnView as my graphics viewer & encoding with Google's pre-built Windows binary), but if I'm understanding my Google searches right (and I might not be because I'm not a programmer), it's because Python (at least under Windows?) doesn't support WebP properly until at least 3.11 & PDF Arranger is using Python 3.10?

**Comment by jeromerobert** [comment]
> it's because Python (at least under Windows?) doesn't support WebP properly

Python itself does not support any image type. Python itself does not support image processing at all. We use PIL and PIL support Webp.

> I do get none/none

Please try to add the webp MIME type in your registry as specified in #937 and it should solve the issue.
